### ISSUE TITLE
Admin should be able to edit mentor details

### ISSUE THREAD

**Issue by anjula-sack** [sef-global/scholarx-backend#168]
**Description:**
- Admin should be able to edit mentor details

**Tasks:**
- We need an endpoint to update mentor details like category, name, profile picture etc

**Acceptance Criteria:**
- There's an endpoint to update mentor details only accessible to admins

**Comment by anjula-sack** [comment]
@Disura-Randunu Would you like to work on this?

**Comment by Disura-Randunu** [comment]
@anjula-sack Sure. Should I prioritize this issue or #155

**Comment by anjula-sack** [comment]
> @anjula-sack Sure. Should I prioritize this issue or #155

You can prioritize this

**Comment by Disura-Randunu** [comment]
@anjula-sack img-url is a property of profile. Shall I make it possible to update other profile data within the mentor update as well?

**Comment by Disura-Randunu** [comment]
@anjula-sack Also, since there's an image, it will be a form-data request. The issue is with type conversions specially with key-value pairs in the "application" property. those values will always be strings since we're passing them through a form data like below.
![Screenshot 2024-09-16 134716](https://github.com/user-attachments/assets/3a94c3f2-43c4-4376-8a5f-1a8318576dbc)

1. we could pass all the other data as a JSON value in the form and parse it in the backend
2. or we could convert in the backend but there will be multiple checks
3. or else make a seperate API call for the image upload and then use the returned string as the imageUrl in another request along with other data. 
4. otherwise can just convert the image to base64

what do you think?

**Comment by anjula-sack** [comment]
Yeah let's go with first option since it's more flexible and we probably will refactor how we are handling the applications. @Disura-Randunu

**Comment by Disura-Randunu** [comment]
@anjula-sack what about the profile data other than image? Should they be updatable within mentor update?

**Comment by anjula-sack** [comment]
> @anjula-sack what about the profile data other than image? Should they be updatable within mentor update?

yes

**Comment by Disura-Randunu** [comment]
@anjula-sack added PR
